### DISCUSSION TITLE
ReferenceError: FormData is not defined at SerializerNativeImpl.serialize

### DISCUSSION THREAD

**Discussion by individual8** [neet/masto.js#976] [Q&A] [answered]
Using the example of uploading media to Mastodon, I get the error `ReferenceError: FormData is not defined at SerializerNativeImpl.serialize` upon execution.

```
try {
    // upload media
    const remoteFile = await fetch(mediaUrl);
    const attachment = await masto.v2.media.create({
      file: remoteFile.blob()
    });
    try {
      // post status
      const status = await masto.v1.statuses.create({
        status: caption,
        visibility: "public",
        mediaIds: [attachment.id],
      });
      // save posted photo to database
      await updatePhotosPosted(service, photosPosted, photoToBePosted);
    } catch (e) {
      functions.logger.log(`Error publishing photo on Mastodon.`, e);
    }
  } catch (e) {
    functions.logger.log(`Error uploading photo to Mastodon.`, e);
  }
```

How can I resolve this issue? I'm using `"masto": "^6.2.0"`, as listed inpackage.json.

**Comment by neet** [comment, marked as answer]
Hi @individual8, thank you for posting a question. I suppose that you're using an unsupported JavaScript environment. Can you verify that you're using Node.js >= 18.0.0?

In principle, we're not supporting Node.js older than 18.x.x because it doesn't offer `FormData`, `fetch` and other APIs, making it difficult to be compatible with browser environments.
